modules/auxiliary/scanner/smb/smb_lookupsid.rb

```diff
@@ -43,270 +44,97 @@ def initialize
       }
+    }
+
+    # Store the domain information
+    report_note(
+      :host => self.simple.peerhost,
+      :proto => 'tcp',
+      :port => self.simple.peerport,
+      :type => 'smb.domain.lookupsid',
+      :data => info[:domain]
+    )
 
-      target_sid = case action.name.upcase
-      when 'LOCAL'
-        host_sid
-      when 'DOMAIN'
-        # Fallthrough to the host SID if no domain SID was returned
-        unless domain_sid
-          print_error("No domain SID identified, falling back to the local SID...")
-        end
-        domain_sid || host_sid
-      end
-
-      min_rid = datastore['MinRID']
-      # Brute force through a common RID range
-
-      min_rid.upto(datastore['MaxRID']) do |rid|
-
-        stub =
-          phandle +
-          NDR.long(1) +
-          NDR.long(rand(0x10000000)) +
-          NDR.long(1) +
-          NDR.long(rand(0x10000000)) +
-          NDR.long(5) +
-          smb_pack_sid(target_sid) +
-          NDR.long(rid) +
-          NDR.long(0) +
-          NDR.long(0) +
-          NDR.long(1) +
-          NDR.long(0)
-
-        dcerpc.call(15, stub)
-        resp = dcerpc.last_response ? dcerpc.last_response.stub_data : nil
-
-        # Skip the "not mapped" error message
-        if(resp and resp[-4,4].unpack("V")[0] == 0xc0000073)
-          next
-        end
-
-        # Stop if we are seeing access denied
-        if(resp and resp[-4,4].unpack("V")[0] == 0xc0000022)
-          break
-        end
-
-        utype,uname = smb_parse_sid_lookup(resp)
-        case utype
-        when 1
-          print_status("USER=#{uname} RID=#{rid}")
-          domain[:users][rid] = uname
-        when 2
-          domain[:groups][rid] = uname
-          print_status("GROUP=#{uname} RID=#{rid}")
-        else
-          print_status("TYPE=#{utype} NAME=#{uname} rid=#{rid}")
+    pipe_info = "PIPE(#{lsarpc_pipe.name})"
+    local_info = "LOCAL(#{info[:local][:name]} - #{info[:local][:sid]})"
+    domain_info = "DOMAIN(#{info[:domain][:name]} - #{info[:domain][:sid]})"
+    all_info = "#{pipe_info} #{local_info} #{domain_info}"
+    print_status(all_info)
+
+    target_sid = case action.name.upcase
+                 when 'LOCAL'
+                   info[:local][:sid] == 'null' ? info[:domain][:sid] : info[:local][:sid]
+                 when 'DOMAIN'
+                   # Fallthrough to the host SID if no domain SID was returned
+                   if info[:domain][:sid] == 'null'
+                     print_error 'No domain SID identified, falling back to the local SID...'
+                     info[:local][:sid]
+                   else
+                     info[:domain][:sid]
+                   end
+                 end
+
+    min_rid = datastore['MinRID']
+    max_rid = datastore['MaxRID']
+
+    # Brute force through a common RID range
+    min_rid.upto(max_rid) do |rid|
+      print "%bld%blu[*]%clr Trying RID #{rid} / #{max_rid}\r"
```

Comment:
It should probably be behind verbose printing logic

Reply:
Well when I run it with the default MinRID and MaxRID options, it takes a few minutes to finish against my DC. If this weren't printed, there wouldn't be anything displayed and it would appear as though the module had hung.